### ISSUE TITLE
model_ops: gemma YAMLs name the out-of-place | as torch.or_, so it is tested as an in-place op

### ISSUE THREAD

**Issue by moriohara** [torch-spyre/torch-spyre#4669]
### What happened?

The two merged gemma model-ops YAMLs record the out-of-place `|` from
[modeling_gemma4.py#L2307](https://github.com/huggingface/transformers/blob/v5.15.0/src/transformers/models/gemma4/modeling_gemma4.py#L2307)
(`multimodal_mask = image_mask | video_mask | audio_mask`) under the op name `torch.or_`:

* `tests/resource/models/gemma-4-12B-it.yaml` lines 24-25, 2562-2563
* `tests/resource/models/gemma-4-26B-A4B-it.yaml` lines 24-25, 5411-5412
* `tests/configs/model_ops_tests/gemma-4-12B-it.yaml` lines 106, 124, 2852, 2870
* `tests/configs/model_ops_tests/gemma-4-26B-A4B-it.yaml` lines 106, 124, 5960, 5978

In this registry a trailing `_` means **in-place** (`torch.add` vs `torch.add_`, plus
`clamp_`, `scatter_`, `copy_`, `masked_fill_`, `index_copy_`), so the adapter implements it
in-place, consistent with the name and wrong for the captured op:

```python
def _tensor_or_(x: torch.Tensor, other):
    x |= other
    return x

    "torch.or_": OpAdapter("torch.or_", _tensor_or_, is_inplace=True),   # op_registry.py:452
```

`is_inplace=True` also makes the harness compare the mutated `arg0` rather than the returned
value (`tests/models/runner.py:422`) — the wrong tensor for an op that does not mutate its
input.

Nothing fails today, which is why this has gone unnoticed. Two reasons, both accidental:

1. the four cases have **equal-shaped** operands (`bool[1, 24]` and `bool[1, 24]`), where
   in-place and out-of-place agree — unlike the granite-embedding mask case in #3825
   (`bool[1,1,13,1] & bool[1,1,1,13]`), which broadcasts and therefore fails outright;
2. both operands are declared `dtype: torch.bool` with no `init:`, so the values are
   degenerate and the comparison is weak either way.

So these cases currently pass while exercising a different operation from the one the model
runs.

### What did you expect to happen?

The out-of-place `|` should use the real torch spelling, leaving `_` to mean in-place:
`torch.bitwise_or`, which is **already registered out-of-place** in `op_registry.py:451`,
directly above the in-place `torch.or_`. That makes the rename a YAML-only change.

Root cause of the name is the generator, tracked in torch-spyre/hf-adapters#546 (#4662 closed as superseded): `_extract_op_name` maps
`_operator.or_` → `torch.or_` and `_operator.ior` → `torch.Tensor.or`, inverting the
convention in both directions (the trailing underscore in `operator.or_` is only Python
keyword-avoidance — `operator.or_.__doc__` is "Same as a | b.").

Scope note: the same naming defect reaches the `torch.and_` cases in the open encoder-model
YAML PRs #3825, #3827, #3828 and #3829, where at least one case broadcasts and therefore
actually fails. This issue covers only the merged gemma YAMLs.

Deliberately **not** part of the fix: removing the now-unused `torch.or_` registry entry.
Until the generator stops emitting the underscored name, a YAML generated with the old
naming would make v1 raise `KeyError` at import and collect **zero** tests for every model
(`test_model_ops.py:76-80` raises where v2 only warns, and
`model_cases_loader.load_all_cases` globs every file in `tests/resource/models/`). Retiring
`torch.or_` / `torch.and_` belongs with the generator fix in torch-spyre/hf-adapters#546.

### How to reproduce

The mismatch is visible without a card — the adapter mutates its first argument, which the
captured `|` does not:

```bash
cd <repo>/tests
TORCH_DEVICE_BACKEND_AUTOLOAD=0 python3 - <<'EOF'
import torch
from models.op_registry import OP_REGISTRY

a = torch.zeros(1, 24, dtype=torch.bool)   # the gemma case's shapes
b = torch.ones(1, 24, dtype=torch.bool)
OP_REGISTRY["torch.or_"].fn(a, b)
print("arg0 clobbered:", bool(a.all()))            # True  -> in-place
print("is_inplace    :", OP_REGISTRY["torch.or_"].is_inplace)

# and where it breaks outright, once operands broadcast (as in #3825's mask case)
c = torch.zeros(1, 1, 13, 1, dtype=torch.bool)
d = torch.ones(1, 1, 1, 13, dtype=torch.bool)
try:
    OP_REGISTRY["torch.or_"].fn(c, d)
except RuntimeError as e:
    print("broadcast     :", e)
EOF
```

### Environment

torch-spyre `fcac2334`, x86_64 + Spyre card. Found while reviewing #3825.
